api/access/server.go

```diff
@@ -130,8 +129,25 @@ func (s *Server) GetBlockByHeight(ctx context.Context, in *access.GetBlockByHeig
 	return nil, errors.New("not implemented")
 }
 
-func (s *Server) GetCollectionByID(ctx context.Context, in *access.GetCollectionByIDRequest) (*access.CollectionResponse, error) {
-	return nil, errors.New("not implemented")
+func (s *Server) GetCollectionByID(_ context.Context, in *access.GetCollectionByIDRequest) (*access.CollectionResponse, error) {
+	id := flow.HashToID(in.Id)
+	lightCollection, err := s.index.Collection(id)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve collection with ID %x: %w", in.Id, err)
+	}
+
+	collection := entities.Collection{
+		Id: in.Id,
+	}
```

Comment:
I think the naming here is also a bit confusing. I think it should just be called `collection` first, and the second one can be a name related to GRPC (`message`, `entity`?).